x/mongo/driver/topology/CMAP_spec_test.go

```diff
@@ -26,6 +33,32 @@ var skippedTestDescriptions = map[string]string{
 	// That behavior is currently required by the "Client.Disconnect" API, so skip the tests.
 	"When a pool is closed, it MUST first destroy all available connections in that pool": "test requires that close does not aggressively close used connections",
 	"must destroy checked in connection if pool has been closed":                          "test requires that close does not aggressively close used connections",
+	// GODRIVER-1826: The load-balancer SDAM error handling test "errors during authentication are
+	// processed" currently asserts that handshake errors trigger events "pool cleared" then
+	// "connection closed". However, the "error during minPoolSize population clears pool" test
+	// asserts that handshake errors trigger events "connection closed" then "pool cleared". The Go
+	// driver uses the same code path for creating all application connections, so those opposing
+	// event orders cannot be satisfied simultaneously.
+	// TODO(DRIVERS-1785): Re-enable this test once the spec test is updated to use the same event
+	// order as the "errors during authentication are processed" load-balancer SDAM spec test.
```

Comment:
The "errors during authentication are processed" test asserts these events:

```
- connectionCreatedEvent: {}
- poolClearedEvent: {}
- connectionClosedEvent:
    reason: error
- connectionCheckOutFailedEvent:
    reason: connectionError
```

Does the pool guarantee a `connectionCheckOutFailedEvent` is emitted after `poolClearedEvent` and `connectionClosedEvent`?

IIUC the [error handling in `createConnections`](https://github.com/mongodb/mongo-go-driver/blob/29d768ed6b8919c666e2362d455e296c9f4e52c3/x/mongo/driver/topology/pool.go#L826-L845) calls `w.tryDeliver(nil, err)` before `handshakeErrFn` (which generates the `poolClearedEvent`) and `removeConnection` (which generates the `connectionClosedEvent`).

Reply:
Good point, the pool does not guarantee the order of `connectionCheckOutFailedEvent` relative to `poolClearedEvent` or `connectionClosedEvent`, so that could be a source of test problems in the future as well. Once we call `w.tryDeliver()`, there are no ordering guarantees between the `checkOut()` goroutine and the `createConnections()` goroutine.

Reply:
I do not see a straightforward workaround. `createConnections` already relies on `tryDeliver` being called before `handshakeErrFn` to deliver the original connection error (rather than a `poolClearedError`) to the caller of `checkOut`.

The "errors during authentication are processed" test is not failing currently. There may be nothing to do now. If it does fail in the future, it may be necessary to skip or ignore the order of events.